include/maliput_sparse/parser/validator.h

```diff
@@ -0,0 +1,115 @@
+// BSD 3-Clause License
+//
+// Copyright (c) 2022, Woven Planet.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "maliput_sparse/parser/parser.h"
+
+namespace maliput_sparse {
+namespace parser {
+
+/// ValidatorOptions struct that contains the options for the Validator.
+struct ValidatorOptions {
+  /// Verifies adjacency for lanes in a segment.
+  bool lane_adjacency{true};
+};
```

Comment:
I agree, we should collapse them. I am thinking about which could be the best way set them. I'd like to keep the options in a enum.

`Validator`'s construct could receive a vector of enum values with the options to check. I will follow that path. However, I must say that passing a struct with booleans is more user-friendly.

Reply:
enum for options sounds right for that.

struct with bool options is ok to set good defaults and is easy to partially modify.

> Validator's construct could receive a vector of enum values with the options to check

I think this is the safest way to achieve what you want in a type-safe and less-error prone manner. You can expose the struct with booleans to the user and make a validator out of it by actually routing the variables to the right enum values. Then, instead of testing for booleans in the constructor you test for specific test cases.

Reply:
I ran into this situation. Context:
1. We want to collapse Error type and Validator options together.
2. We want to differentiate LogcalLaneAdjacency errors from GeometricalLaneAdjacency errors.

The thing is that it does not make that much of sense to be able to select just geometricalLaneAdjacency validation. Why? Well, because the selection of which lanes should be checked for adjacency are determined via the graph of lanes, and for such a thing we need to have a correct structure of lanes in the segment logically validated. (Checking geometrical adjacency without checking logical adjacency first makes it impossible to check the former)

Then having different collections for errors and options like:

```cpp

struct ValidatorOptions {
  /// Verifies adjacency for lanes in a segment.
  bool lane_adjacency{true};
};
```
```cpp

class Validator {
 public:
  /// Error struct that contains the error message, type, and severity.
  struct Error {
    /// The type of error.
    enum class Type {
      kLogicalLaneAdjacency,
      kGeometricalLaneAdjacency,
    };
    ....
```

still makes sense. I am still thinking on another probable usecase for supporting this code.

Reply:
Understood, the problem is that you have a prerequisite then. You don't have support for that with this type of configuration. Maybe, enabling `kGeometricalLaneAdjacency` enables `kLogicalLaneAdjacency` as it a prerequisite?

Reply:
Done. ptal